src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -49,9 +49,13 @@ public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
             {
                 yamlDocument = LoadYamlDocument(input);
             }
+            catch (SemanticErrorException ex) {
+                diagnostic.Errors.Add(new OpenApiError($"#char={ex.Start.Index},{ex.End.Index}", ex.Message));
+                return new OpenApiDocument();
+            }
             catch (SyntaxErrorException ex)
```

Comment:
> catch (SyntaxErrorException ex) [](start = 11, length = 32)

Should we catch the generic YamlException as well? I don't know if we are missing any others?

Reply:
For example, [https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/Core/AnchorNotFoundException.cs](https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/Core/AnchorNotFoundException.cs), [https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/Core/MaximumRecursionLevelReachedException.cs](https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/Core/MaximumRecursionLevelReachedException.cs)

---
In reply to: [194829393](https://github.com/Microsoft/OpenAPI.NET/pull/276#discussion_r194829393) [](ancestors = 194829393)

Reply:
Well, I guess there's nothing else in SharpYaml [https://github.com/xoofx/SharpYaml/tree/master/SharpYaml](https://github.com/xoofx/SharpYaml/tree/master/SharpYaml), but there are more exceptions in YamlDotNet (the original repo). 

I still think it makes sense to catch the more generic exception though :)

---
In reply to: [194829393](https://github.com/Microsoft/OpenAPI.NET/pull/276#discussion_r194829393) [](ancestors = 194829393)

Reply:
Yeah, I can catch the generic YamlException too.

---
In reply to: [194830182](https://github.com/Microsoft/OpenAPI.NET/pull/276#discussion_r194830182) [](ancestors = 194830182,194829393)